corehq/motech/repeaters/models.py

```diff
@@ -196,23 +196,26 @@ def all_ready(self):
                 .filter(repeat_records_ready_to_send))
 
 
-class RepeaterStub(models.Model):
-    """
-    This model links the SQLRepeatRecords of a Repeater.
-    """
+class SQLRepeater(models.Model):
     domain = models.CharField(max_length=126)
     repeater_id = models.CharField(max_length=36)
     is_paused = models.BooleanField(default=False)
     next_attempt_at = models.DateTimeField(null=True, blank=True)
     last_attempt_at = models.DateTimeField(null=True, blank=True)
 
-    objects = RepeaterStubManager()
+    objects = RepeaterManager()
+
+    connection_settings = models.ForeignKey(
+        ConnectionSettings,
+        on_delete=models.PROTECT
```

Comment:
In my opinion, `PROTECT` strategy should be optimum here. But are there any use cases where we want to allow that ConnectionSettings instance referring to a Repeater to be deleted?

Reply:
You are right. We don't want ConnectionSettings for a Repeater to be deleted before the Repeater is deleted.